### PR TITLE
Add explicit instantiation for complex RankFourTensor operation

### LINKED ISSUE #15734: Add explicit instantiation for RankFourTensor complex operation
## Reason
Spectral solver in Magpie requires the use of complex tensor operations. See https://github.com/idaholab/magpie/pull/441

## Design
Simple explicit instantiation, more may be needed.

## Impact
Will allow for the development of spectral solvers.

### PR DESCRIPTION
**Pull request by recuero**
Closes #15734

### CODE REVIEW

File: framework/src/utils/RankFourTensor.C [hunk 1]
```diff
@@ -12,6 +12,8 @@
 
 template class RankFourTensorTempl<Real>;
 template class RankFourTensorTempl<DualReal>;
+template class RankTwoTensorTempl<Complex>;
+template class RankFourTensorTempl<Complex>;
```
Comment (dschwen) [inline framework/src/utils/RankFourTensor.C:16]: The  `Complex` typedef is not known in MOOSE. This instantiation should be in Magpie for now.

Comment (dschwen) [reply framework/src/utils/RankFourTensor.C:16]: In [`ComplexTypes.C`](https://github.com/idaholab/magpie/blob/devel/src/utils/ComplexTypes.C) to be precise

Comment (recuero) [reply framework/src/utils/RankFourTensor.C:16]: Can't `Complex` be available in MOOSE through libmesh: `libmesh_common.h`?

Comment (dschwen) [reply framework/src/utils/RankFourTensor.C]: Maybe, if libmesh was compiled with complex number support. But that's a bit besides the point. By design I'm trying to keep unnecessary explicit instantiations for complex types out of the main moose code base. Everything else related to complex types is in Magpie, so this belongs there, too. Maybe at some point in the future we'll have proper complex number support for the FEM parts of MOOSE and these instantiations will be useful in MOOSE, but I don't think thats the case quite yet.

Comment (recuero) [reply framework/src/utils/RankFourTensor.C]: > if libmesh was compiled with complex number support.

I think that typedef is there all the time. I understood your point.

**Comment by recuero** [comment]
This can be done in Magpie

**Pull request closed without merging**